### ISSUE TITLE
Implement a job queue for the server side 

### ISSUE THREAD

**Issue by adri9valle** [wazuh/wazuh-splunk#581]
It's time to create a new module that accepts jobs, then it puts them into a job queue. Once a job is finished we must send a push notification to the frontend side where we show the result (success|fail)

Tasks:

- [x] Implement the module (backend)

**Comment by adri9valle** [comment]
Working on this branch: [3.9-queue](https://github.com/wazuh/wazuh-splunk/tree/3.9-queue)

**Comment by adri9valle** [comment]
Today's progress:
- Add input stanza for executing each minute the script that checks the jobs queue.
- Add the script that will consult the jobs saved on the kvStore.
- Trying to insert, remove and get jobs on the kvStore.

**Comment by adri9valle** [comment]
Today's progress:

I have tried several ways to implement a job queue, trying to save in an array on memory and read this array every 60 seconds with threads, crontab, etc. 

Finally, I'll use the kvStore like at the beginning, this has several advantages, if Splunk has been restart the jobs in the queue won't be removed, also I have added several extra fields like the date when was the job added and the field `done` to check if this job was done or not.

To run each minute the script that will check the jobs queue I have used the `inputs.conf` file exec the script.

Abilities implemented at the moment:

- Add jobs
- Get the jobs
- Filter undone jobs

**Comment by adri9valle** [comment]
Today's progress:

**Job finished.**

I have finished the job queue.  I've implemented two ways to mark as done the job. Once is when the job was executed without fails, in the kvStore the job is updated and the field `done` is set to `True`, in the next check this job will be ignored and we will have a history of the jobs executed. The other way and is when the job is executed is deleted from the kvStore, so the next check won't contain this job and won't be executed again. Currently, the jobs are deleted after have been executed, but maybe in the future, we will need to have a track of the executed jobs so we only need to change the way to mark the job as done.

Regards,

Solved: https://github.com/wazuh/wazuh-splunk/pull/629
